### ISSUE TITLE
Auto id with prefix

### ISSUE THREAD

**Issue by sahanh** [fuel/core#789]
Fieldset_Field 

if ($form->get_config('auto_id', false) === true and $this->get_attribute('id') == '')

this code overrides auto id addition of Form_Instance.

(Form_Instance class code $attributes['id'] = $this->get_config('auto_id_prefix', 'form_').$attributes['name'];) 
This code doesn't add the auto_id_prefix and it ignores auto_id generation (which adds the prefix) because Fielsset_Field add an id (without prefix)

**Comment by WanWizard** [comment]
You mean that when Fieldset_Field detects there is no ID set, and it autogenerates an ID, if doesn't add the prefix?

**Comment by sahanh** [comment]
Fieldset_Field adds an id if an id is not set, but it doesn't add the prefix. However Form_Instance has a code already, which does the job. That code works if I comment out the code in Fieldset_Field.

**Comment by jschreuder** [comment]
Don't have time to look into this now and may forget later, but I think I fixed this in develop...

**Comment by mikebranderhorst** [comment]
In 1.1 dev:

config form.php
`auto_id = true`
`auto_id_prefix = form_`
(custom templates used)

echo \Fieldset::forge()->add_model('Model_Test')->build();
Notice the missing form_ for input...

```
    <div class="control-group">
        <label for="form_slug">slug</label>
        <div class="controls">
            <input type="text" id="slug" name="slug" value="" />
        </div>
    </div>
```

class field.php
comment line 392

```
        //  $this->set_attribute('id', $auto_id);
```

Notice the missing name for label...

```
    <div class="control-group">
        <label for="form_">slug</label>
        <div class="controls">
            <input type="text" name="slug" value="" id="form_slug" />
        </div>
    </div>
```

It seems that label needs the form_ + name set at field.php line 392
but input is set at form instance.php line 221

So when in field.php the id=name isset then in instance.php $attribute[''id'] is not empty anymore thus the prefix form_ is not prefixed.

I could not come up with a solution, sorry...

**Comment by sahanh** [comment]
I think I found the issue, and submitted a commit. https://github.com/sahanz/core/commit/5617041e0ed49799b24c3f7be7e0f467140ca3f4

**Comment by sahanh** [comment]
I think this time I got it right, https://github.com/fuel/core/pull/819

**Comment by jschreuder** [comment]
Closed, will be handled by issue #839.

**Comment by jschreuder** [comment]
This is fixed now by aa197e81bc0fe6ebaaf732747ce6160d3500229f
